x-pack/plugins/lens/public/trigger_actions/types.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { apiIsOfType } from '@kbn/presentation-publishing';
+import { HasLensConfig, ViewUnderlyingDataArgs } from '../embeddable';
+
+export type OpenInDiscoverActionApi = HasLensConfig & {
```

Comment:
Thanks. I've seen the `LensApi` changes you are making in https://github.com/elastic/kibana/pull/176869/files#diff-cf9423bd506b04c1decd8629e55629bdd19b74ad59f869f3abe994c1d364b2bc and Option 1 makes sense.

Option 2 seems like it would also work, but it seems like an unnecessary abstraction.